files/setup-010-veba-ui.sh

```diff
@@ -0,0 +1,114 @@
+#!/bin/bash
+# Copyright 2019 VMware, Inc. All rights reserved.
+# SPDX-License-Identifier: BSD-2
+
+# Setup Knative UI
+
+set -euo pipefail
+
+echo -e "\e[92mSetting up VEBA UI RBAC ..." > /dev/console
+cat > /root/config/veba-ui-rbac.yaml << EOF
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: veba-ui
+  namespace: vmware-system
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: veba-ui
+  namespace: vmware-functions
+rules:
+  - apiGroups: [""]
+    resources: ["secrets"]
+    verbs: ["watch", "get", "list", "create", "update", "delete"]
+  - apiGroups: ["serving.knative.dev"]
+    resources: ["services"]
+    verbs: ["watch", "get", "list", "create", "update", "delete"]
+  - apiGroups: ["eventing.knative.dev"]
+    resources: ["triggers"]
+    verbs: ["watch", get", "list", "create", "update", "delete"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: veba-ui
+  namespace: vmware-functions
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: veba-ui
+subjects:
+- kind: ServiceAccount
+  name: veba-ui
+  namespace: vmware-system
+EOF
+
+kubectl apply -f /root/config/veba-ui-rbac.yaml
+
+ESCAPED_VCENTER_SERVER=$(echo -n ${VCENTER_SERVER} | python -c 'import sys,json;data=sys.stdin.read(); print json.dumps(data)[1:-1]')
+ESCAPED_VCENTER_USERNAME_FOR_VEBA_UI=$(echo -n ${VCENTER_USERNAME_FOR_VEBA_UI} | python -c 'import sys,json;data=sys.stdin.read(); print json.dumps(data)[1:-1]')
+ESCAPED_VCENTER_PASSWORD_FOR_VEBA_UI=$(echo -n ${VCENTER_PASSWORD_FOR_VEBA_UI} | python -c 'import sys,json;data=sys.stdin.read(); print json.dumps(data)[1:-1]')
+
+echo -e "\e[92mSetting up VEBA UI Secret ..." > /dev/console
+kubectl -n vmware-system create secret generic veba-ui-secret \
+    --from-literal=VCENTER_FQDN=${ESCAPED_VCENTER_SERVER} \
+    --from-literal=VCENTER_PORT=443 \
```

Comment:
QQ: so we always assume https and that fixed port? Is this also true for general VEBA VC setup? If so, good with me for now.

Reply:
That's correct. I had thought about exposing this through an OVF property but realized we don't offer this configuration for VMware Event Router and just add additional input (even if its set to default). We can revisit if we need to support further customization